### PR TITLE
Documented CLI arguments for upgrade script

### PR DESCRIPTION
**Pull request by stuclem**
Fixes https://github.com/vmware/vic-product/issues/1608

@andrewtchin and @morris-jason can you please review? Thanks!

### CODE REVIEW

File: docs/user_doc/vic_vsphere_admin/upgrade_appliance.md [hunk 1]
```diff
@@ -85,3 +87,113 @@ After you see confirmation that the upgrade has completed successfully, the upgr
...
+          <td><code>--manual-disks</code></td>
+          <td>Skip the automated  disk migration. Use this option if you  manually moved the disks from the old appliance to the new appliance. * </td>
+        </tr>
+      </table>
+
+&#42; To use the `--manual-disks` option, follow the instructions in [Upgrade the vSphere Integrated Containers Appliance by Manually Moving Disks](upgrade_appliance_manual.md).
+
+### Example: Upgrade Appliance with Embedded Platform Services Controller
+
+The following command upgrades a vSphere Integrated Containers 1.3.1 appliance. The new appliance runs in a vCenter Server instance with an embedded Platform Services Controller. The old appliance is removed when the upgrade finishes.
```
Comment (andrewtchin) [inline docs/user_doc/vic_vsphere_admin/upgrade_appliance.md]: The destroy flag also still requires interactive confirmation y/n even if all other values are provided. I'm not sure if this warrants a note.

File: docs/user_doc/vic_vsphere_admin/upgrade_appliance.md [hunk 2]
```diff
@@ -85,3 +87,113 @@ After you see confirmation that the upgrade has completed successfully, the upgr
...
+          <td><code>--username</code></td>
+          <td> Specify the Single Sign-On user name of a vSphere administrator account.</td>
+        </tr>
+        <tr>
+          <td><code>--password</code></td>
+          <td>Specify the Single Sign-On password of a vSphere administrator account.</td>
+        </tr>
+        <tr>
+          <td><code>--fingerprint</code></td>
+          <td>Specify the thumbprint of the vCenter Server certificate. Use upper-case letters and colon delimitation in the thumbprint. Do not use space delimitation.</td>
```
Comment (andrewtchin) [inline docs/user_doc/vic_vsphere_admin/upgrade_appliance.md]: This requires a specific format containing the target IP addr obtained by using `govc`. Here's a full command 
```
root@Photon [ ~ ]# /etc/vmware/upgrade/upgrade.sh --target 10.160.140.175 --password 'Admin!23' --username 'administrator@vsphere.local' --dc ha-datacenter --fingerprint '10.160.140.175 49:8C:56:6B:F0:E6:54:D1:3F:77:4A:81:DE:BD:61:8B:80:CE:DF:E6' --appliance-username root --appliance-password password --appliance-target 10.192.38.196 --appliance-version v1.3.1 --destroy --ssh-insecure-skip-verify --embedded-psc
```

File: docs/user_doc/vic_vsphere_admin/upgrade_appliance.md [hunk 3]
```diff
@@ -85,3 +87,113 @@ After you see confirmation that the upgrade has completed successfully, the upgr
...
+<pre>./upgrade.sh --target <i>new_appliance_address</i>
+--username 'Administrator@vsphere.local'
+--password 'P@ssW0rd!'
+--fingerprint <i>vcenter_server_thumbrint</i>
+--dc oldApplianceDatacenter
+--embedded-psc
+--appliance-target <i>old_appliance_address</i>
+--appliance-username root
+--appliance-password <i>old_appliance_root_password</i>
+--appliance-version 1.3.1
```
Comment (andrewtchin) [inline docs/user_doc/vic_vsphere_admin/upgrade_appliance.md]: version must contain the `v` like in the help text

**Comment by andrewtchin** [comment]
Also upgrade related - I just confirmed that the auto disk copy works between ESXs in the same DC

**Comment by andrewtchin** [comment]
Merged to release branch in https://github.com/vmware/vic-product/pull/1678

File: docs/user_doc/vic_vsphere_admin/upgrade_appliance.md (hunk 1 above)
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/upgrade_appliance.md]: I don't think so here, but I added this detail in the table of options.

File: docs/user_doc/vic_vsphere_admin/upgrade_appliance.md (hunk 3 above)
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/upgrade_appliance.md]: Added: "Specify the IP address of vCenter Server and the thumbprint of the vCenter Server certificate, in the format <code>--fingerprint '<i>vcenter_server_address</i> <i>vcenter_server_thumbrint</i>'</code>. Use upper-case letters and colon delimitation in the thumbprint. Do not use space delimitation." Updated the examples.

File: docs/user_doc/vic_vsphere_admin/upgrade_appliance.md (hunk 2 above)
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/upgrade_appliance.md]: Added: "Specify the IP address of vCenter Server and the thumbprint of the vCenter Server certificate, in the format <code>--fingerprint '<i>vcenter_server_address</i> <i>vcenter_server_thumbrint</i>'</code>. Use upper-case letters and colon delimitation in the thumbprint. Do not use space delimitation." Updated the examples.

File: docs/user_doc/vic_vsphere_admin/upgrade_appliance.md (hunk 3 above)
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/upgrade_appliance.md]: Done

File: docs/user_doc/vic_vsphere_admin/upgrade_appliance.md [hunk 4]
```diff
@@ -85,3 +87,113 @@ After you see confirmation that the upgrade has completed successfully, the upgr
...
+        <tr>
+          <td><code>--appliance-username</code></td>
+          <td> Specify the user name for the old appliance, usually  <code>root</code>.</td>
+        </tr>
+        <tr>
+          <td><code>--appliance-password</code></td>
+          <td>Specify the password for the <code>root</code> user on the old appliance.</td>
+        </tr>
+        <tr>
+          <td><code>--ssh-insecure-skip-verify</code></td>
```
[the remark this answers is not in the document]
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/upgrade_appliance.md]: Done

**Comment by stuclem** [comment]
Thanks @andrewtchin. Regarding "Also upgrade related - I just confirmed that the auto disk copy works between ESXs in the same DC", do you mean that it ONLY works if the hosts are in the same DC?

File: docs/user_doc/vic_vsphere_admin/upgrade_appliance.md [hunk 5]
```diff
@@ -85,3 +87,113 @@ After you see confirmation that the upgrade has completed successfully, the upgr
 **Troubleshooting**
 
 If upgrade fails, generate a log bundle and obtain the upgrade log to provide to VMware support. For information about obtaining the logs, see [Access and Configure Appliance Logs](appliance_logs.md).
+
+## Specify Command Line Options During Appliance Upgrade <a id="upgradeoptions"></a>
```
[the remark this answers is not in the document]
Comment (stuclem) [reply docs/user_doc/vic_vsphere_admin/upgrade_appliance.md]: Done in https://github.com/vmware/vic-product/pull/1650/commits/715608938a5e219f04f81ad39d68a15c18fc5d29

**Comment by andrewtchin** [comment]
So technically it should work cross cluster within a DC, but I'd be more comfortable saying we support cross host within a cluster since that's what we've tested so far.

**Comment by stuclem** [comment]
@andrewtchin I added a prereq: "The upgrade process copies data from the old appliance to the new appliance. Consequently, if you deployed the appliances to a cluster, the virtual disks for the two appliances must be located in the same datastore cluster." 

This is in https://github.com/vmware/vic-product/pull/1650/commits/e3160d017adefb412479b63838d0bf584449383d.

**Comment by andrewtchin** [comment]
LGTM

**Pull request merged**